citadel_workspace_service/tests/file_transfer.rs

```diff
@@ -0,0 +1,898 @@
+                                let cmp = include_bytes!("../../../resources/test.txt");
+                                let streamed_data =
+                                    tokio::fs::read(path.clone().unwrap()).await.unwrap();
+                                assert_eq!(
+                                    cmp,
+                                    streamed_data.as_slice(),
+                                    "Original data and streamed data does not match"
+                                );
+                            }
+
+                            ObjectTransferStatus::ReceptionBeginning(file_path, vfm) => {
+                                path = Some(file_path);
+                                assert_eq!(vfm.get_target_name(), "test.txt")
+                            }
+
+                            _ => {}
+                        }
+                    }
+                } else {
+                    panic!("File Transfer Failed on data stream");
+                }*/
+            } else {
+                panic!("File Transfer P2P Failure");
+            }
+        } else {
+            panic!("File Transfer Request failed: {deserialized_service_a_payload_response:?}");
+        }
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_citadel_workspace_service_c2s_revfs() -> Result<(), Box<dyn Error>> {
+        citadel_logging::setup_log();
+        info!(target: "citadel", "above server spawn");
+        let bind_address_internal_service: SocketAddr = "127.0.0.1:55518".parse().unwrap();
+
+        // TCP client (GUI, CLI) -> Internal Service -> Receiver File Transfer Kernel server
+        let server_success = &Arc::new(AtomicBool::new(false));
+        let (server, server_bind_address) = server_info_file_transfer(server_success.clone());
+
+        tokio::task::spawn(server);
+
+        info!(target: "citadel", "sub server spawn");
+        let internal_service_kernel = CitadelWorkspaceService::new(bind_address_internal_service);
+        let internal_service = NodeBuilder::default()
+            .with_node_type(NodeType::Peer)
+            .with_backend(BackendType::InMemory)
+            .build(internal_service_kernel)?;
+
+        tokio::task::spawn(internal_service);
+
+        // give time for both the server and internal service to run
+
+        tokio::time::sleep(Duration::from_millis(2000)).await;
+
+        info!(target: "citadel", "about to connect to internal service");
+
+        let (to_service, mut from_service, uuid, cid) = register_and_connect_to_server(
+            bind_address_internal_service,
+            server_bind_address,
+            "John Doe",
+            "john.doe",
+            "secret",
+        )
+        .await
+        .unwrap();
+
+        // Push file to REVFS
+        let file_to_send = PathBuf::from("../../resources/test.txt");
+        let virtual_path = PathBuf::from("vfs/");
```

Comment:
Should be the target output path, e.g:
```suggestion
        let virtual_path = PathBuf::from("/test.txt");
```